### ISSUE TITLE
WaveformDragMode seek

### ISSUE THREAD

**Issue by jason30704** [bbc/peaks.js#539]
Hello,
WaveformDragMode currently only has 'scroll' and 'insert-segment'. But this will make zoomview unable to seek. Is it possible for ZoomView to retain the ability to seek even when scrolling is disabled?

**Comment by chrisn** [comment]
Currently, to seek in the zoomview you can either click in the waveform or drag the playhead, so this should still work if scrolling is disabled.

**Comment by jason30704** [comment]
But if I want to prevent zoomview scroll now, can't I only set Insert segment mode?
This means that if you want zoomview to be unable to scroll, it will only be in Insert segment mode. Therefore, it will not be able to seek.
Or maybe I misunderstood some part.
thank you.

**Comment by chrisn** [comment]
Sorry, I think that's right, there currently isn't a way to only disable scrolling.

**Comment by chrisn** [comment]
I'm not sure I understand what you want to be able to do. Do you want an option to disable scrolling, but also not insert segments?

**Comment by jason30704** [comment]
Yes, I hope that zoomview can disable scrolling and still retain the ability to seek when clicking.
But currently using insert-segments, I will not be able to seek by clicking.

**Comment by chrisn** [comment]
Thanks, so we could add an option and method to enable and disable scrolling. But we'll need to think about how this would interact with the existing `autoScroll` option and `enableAutoScroll(boolean, options)` method.

**Comment by chrisn** [comment]
I agree, disabling scroll would also disable auto-scroll. I'm thinking about the API we might provide. The simplest is to add a new `enableScroll(boolean)` method, and no config option, so the default is scroll enabled.

I thought about combining the auto-scroll controls into a single `enableScroll(boolean, options)` but this is probably more confusing:

```js
view.enableScroll(false); // Disables all scrolling
view.enableScroll(true); // Enables scroll and auto-scroll
view.enableScroll(true, { auto: false }); // Enables scroll but not auto-scroll
                                          // same as existing view.enableAutoScroll(false)
view.enableScroll(true, { offset: pixels }); // Enables scroll and auto-scroll, sets auto-scroll offset
```

**Comment by jason30704** [comment]
Based on my current experience, I agree that separate settings may be more suitable for current project usage.
